lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1388,8 +1388,22 @@ private FoundHits queryHits(
                                                                       sr.ascending ? "asc" : "desc"));
             sort.add(sortOrderWrapper);
         }
-        String indexName = objectType == null ? getAnyIndexPattern() : checkIndex(objectType);
-        String urlPath = "/" + indexName + "/" + getDataTableName() + "/_search";
+
+        StringBuffer indexName = new StringBuffer();
+        for(int ii=0; ii<objectTypes.size(); ii++) {
+
+            String objectType = objectTypes.get(ii);
+
+            indexName.append(
+                    objectType == null ? getAnyIndexPattern() : checkIndex(objectType)
```

Comment:
I don't get this. If there's a null entry in the list of types, it adds a search term that matches all types to the url? That means if there's a null in the list, there's no point in doing anything other than making the url the result of getAnyIndexPattern() since that's a completely open search re types.

It makes more sense to me to throw an exception if there are any null entries in the type list (Utils.noNulls()), and if the type list is null or empty then the url is getAnyIndexPattern(), otherwise the url is constructed from the type strings via checkIndex().

Reply:
I figured that if there was a null in the list, it would just end up relaxing the search to an unconstrained search. Anyhow I have added object type validation.

Reply:
This API still seems weird to me. If I want to search across all types, it seems intuitive to me to just not include a type list at all or pass an empty list. Passing a list with a single null element is not what I would expect at all.

Reply:
Yes it is weird. I was trying to make minimum changes to get this out fast. It is now fixed. Empty list implies anyType. null in list in invalid.